pyqg/layered_model.py

```diff
@@ -275,48 +275,75 @@ def _initialize_model_diagnostics(self):
         self.add_diagnostic('entspec',
                 description='barotropic enstrophy spectrum',
                 function= (lambda self:
-                    np.abs((self.Hi[:,np.newaxis,np.newaxis]*self.qh).sum(axis=0))**2/self.H) )
+                    np.abs((self.Hi[:,np.newaxis,np.newaxis]*self.qh).sum(axis=0))**2/self.H),
+                units='m^2/s^2',
+                dims=('l','k')
+        )
 
         self.add_diagnostic('KEspec_modal',
-                description='modal KE spectra',
+                description='modal kinetic energy spectra',
                 function= (lambda self:
-                    self.wv2*(np.abs(self.phn)**2)/self.M**2 ))
+                    self.wv2*(np.abs(self.phn)**2)/self.M**2 ),
+                units='J',
```

Comment:
These units are not right. We need to look at the mathematical expression (the function) to understand what the units should be.

- It starts with `phn`, the modal project of `ph`, which should have the same usits as `ph`: `m2 s-1`.
- We square that: -> `m4 s-2`
- Divide by `M**2`: M has no units, just the number of points in the domain (a normalization)
- Multiply by `wv2` (the wavenumber squared; units `m-2`), so we get to `m2 s-2`

In fluids, we always talk about energy _density_, which is technically `J m-3`, but for boussinesq fluids, we always factor out the density, so divide energy by rho (units `kg m^-3`) which brings us to `m2 s-2` (`J = N s-1 = kg m2 s-2`)